modules/scala3-graal/src/main/scala/scala/cli/graal/BytecodeProcessor.scala

```diff
@@ -0,0 +1,253 @@
+package scala.cli.graal
+
+import org.objectweb.asm._
+
+import java.io.{File, InputStream}
+import java.nio.file.{Files, Path, StandardOpenOption}
+import java.util.jar.{Attributes, JarEntry, JarFile, JarOutputStream, Manifest}
+
+import scala.jdk.CollectionConverters._
+
+trait JarCache {
```

Comment:
Put those extra classes (`JarCache`, `ClassPathEntry`, `TempCache`) in their own file?